test/Microsoft.AspNetCore.Razor.Runtime.Test/Runtime/TagHelpers/TagHelperExecutionContextTest.cs

```diff
@@ -270,51 +273,6 @@ public static TheoryData<HtmlEncoder> HtmlEncoderDataLessNull
             Assert.NotSame(content1, content2);
         }
 
-        public static TheoryData<string, string> DictionaryCaseTestingData
-        {
-            get
-            {
-                return new TheoryData<string, string>
-                {
-                    { "class", "CLaSS" },
-                    { "Class", "class" },
-                    { "Class", "claSS" }
-                };
-            }
-        }
-
-        [Theory]
-        [MemberData(nameof(DictionaryCaseTestingData))]
-        public void HtmlAttributes_IgnoresCase(string originalName, string updatedName)
-        {
-            // Arrange
-            var executionContext = new TagHelperExecutionContext("p", TagMode.StartTagAndEndTag);
-            executionContext.HtmlAttributes.SetAttribute(originalName, "hello");
-
-            // Act
-            executionContext.HtmlAttributes.SetAttribute(updatedName, "something else");
-
-            // Assert
-            var attribute = Assert.Single(executionContext.HtmlAttributes);
-            Assert.Equal(new TagHelperAttribute(originalName, "something else"), attribute);
-        }
-
-        [Theory]
-        [MemberData(nameof(DictionaryCaseTestingData))]
-        public void AllAttributes_IgnoresCase(string originalName, string updatedName)
-        {
-            // Arrange
-            var executionContext = new TagHelperExecutionContext("p", tagMode: TagMode.StartTagAndEndTag);
-            executionContext.AllAttributes.SetAttribute(originalName, value: false);
-
-            // Act
-            executionContext.AllAttributes.SetAttribute(updatedName, true);
-
-            // Assert
-            var attribute = Assert.Single(executionContext.AllAttributes);
-            Assert.Equal(new TagHelperAttribute(originalName, true), attribute);
-        }
```

Comment:
Restore these tests because the scenarios aren't covered otherwise. Either expose `AllAttributes` and `HtmlAttributes` as `// Internal for testing` or create a `TagHelperContext` / `TagHelperOutput` and check what that contains.